Create new help categories related to opening compose box

Niloth-p opened the pull request:
### What does this PR do, and why?
Groups key bindings from message actions into the new categories:
- Compose: New Message
- Compose: Replies

And in the 2nd commit, suggests renaming the "Composing a message" category to "Writing a message" category, to improve clarity, in contrast with the newly added categories.

### External discussion & connections
- [x] Discussed in **#zulip-terminal** in `Re-Categorization of Hotkeys`
- [ ] Fully fixes #
- [ ] Partially fixes issue #
- [ ] Builds upon previous unmerged work in PR #
- [ ] Is a follow-up to work in PR #
- [ ] Requires merge of PR #
- [x] Merge will enable work on #1484

### How did you test this?
- [ ] Manually - Behavioral changes
- [x] Manually - Visual changes
- [ ] Adapting existing automated tests
- [ ] Adding automated tests for new behavior (or missing tests)
- [ ] Existing automated tests should already cover this (*only a refactor of tested code*)

### Self-review checklist for each commit
- [x] It is a [minimal coherent idea](https://github.com/zulip/zulip-terminal#structuring-commits---speeding-up-reviews-merging--development)
- [x] It has a commit summary following the [documented style](https://github.com/zulip/zulip-terminal#structuring-commits---speeding-up-reviews-merging--development) (title & body)
- [x] It has a commit summary describing the  motivation and reasoning for the change
- [x] It individually passes linting and tests
- [ ] It contains test additions for any new behavior
- [ ] It flows clearly from a previous branch commit, and/or prepares for the next commit

### Visual Changes
![image](https://github.com/zulip/zulip-terminal/assets/20315308/6b38c06e-b926-4d0a-8fa6-29b4bf35de08)

zormit reviewed (approved):
This is nice, thank you. I'm looking forward to having a reduced list of "message actions" (which was too long).

Niloth-p commented:
Rebased, resolving merge conflicts and updating INSERT_NEW_LINE's category along with the rest in the 2nd commit.

neiljp commented:
@Niloth-p Thanks for sticking with these improvements - I agree they are definitely helping :+1:

Following on from my summary in #1524, my main concern here is that the categories are small (3 or 4 entries), plus the draft option doesn't entirely fit as a 'new message' (which would make it 2, and find somewhere to put the draft entry?). While it potentially makes it easier to find some entries when there are very many categories, the vertical spacing starts to take over. Do you agree?

I think we'll still gain from splitting these commands out into one section, and the renaming of the other section certainly remains worthwhile too. 

Having just merged the GO_BACK splitting PR (#1514), it also prompts me to think whether the exiting of compose could be included here (too), as a 'Starting/Exiting compose' sort of category. However, that introduces multi-category commands again, and could make naming the single category more challenging - this isn't as important as getting the splitting and renaming done and merged.

Niloth-p commented:
Updates:
- Merged the 2 new categories.
- Rebased onto main, and updated the recently newly added EXIT_COMPOSE.

Things to check:
- The naming of the newly formed category - both internally and externally
- Is the ordering of help keys in the new category satisfactory?

Niloth-p commented:
> While it potentially makes it easier to find some entries when there are very many categories, the vertical spacing starts to take over. Do you agree?

Regarding my opinion,
The spacing would definitely dominate in smaller screens. 3 is quite a small category indeed.

But, I like smaller help categories in general for any application, but that's probably because I try to memorize them (and the related commands) instead of just referring to a single command.
So, tighter categories help in that aspect, instead of me having to break them down into further tiny categories in my own mind, it's already done for me. So, the tightness/fit of entries into categories would be more preferred over the size of categories. 
But, that's probably me, I have no idea what users majorly prefer.

zormit commented:
I think small categories can be nice as long as the titles are really really clear.

My reason would be a bit different than yours: If I'm looking for a specific command I can just skim the headlines and hopefully quickly find the right category. In that category it then should also be quite fast to identify the command I'd be looking for, because there are not that many present.

neiljp commented:
@Niloth-p Thanks for this and your slight update :+1: It's definitely nuanced between the two but let's move on and get this merged for now, and we can always return to examine the category splits later, particularly if we have the section jump keys :)
